### PR TITLE
Category component & search

### PR DESCRIPTION
**Pull request by dsuchy**

### CODE REVIEW

**Review by syphzur — CHANGES_REQUESTED** [review]
please add some functionality (filtering products by category name)

**Review by syphzur — DISMISSED** [review]
looks ok

File: PhotoPlusFrontend/PhotoPlus/src/app/components/category/category.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      this.productsToShow = [];
+      this.products.forEach(product => {
+        console.log(categoryName, product);
+        if (categoryName == product.category) {
+          this.productsToShow.push(product);
+        }
+      });
+    }
+
+  addToCart(product: Product) {
```
Comment (syphzur) [inline PhotoPlusFrontend/PhotoPlus/src/app/components/category/category.component.ts:43]: I think you should move addToCart function to productService. We should avoid duplicate code.

**Review by syphzur — CHANGES_REQUESTED** [review]
fix duplicate code

File: PhotoPlusFrontend/PhotoPlus/src/app/components/category/category.component.ts (hunk 1 above)
Comment (syphzur) [reply PhotoPlusFrontend/PhotoPlus/src/app/components/category/category.component.ts:43]: It is also used in home component.

**Review by syphzur — APPROVED** [review]

File: PhotoPlusFrontend/PhotoPlus/src/app/components/category/category.component.ts (hunk 1 above)
Comment (Suchar230) [reply PhotoPlusFrontend/PhotoPlus/src/app/components/category/category.component.ts:43]: N O

**Pull request merged**